src/terms/operators.jl

```diff
@@ -114,16 +114,23 @@ struct MagneticFieldOperator{T <: Real, AT} <: RealFourierOperator
     kpoint::Kpoint{T}
     Apot::AT  # Apot[α][i,j,k] is the A field in (cartesian) direction α
 end
+
 function apply!(Hψ, op::MagneticFieldOperator, ψ)
     # TODO this could probably be better optimized
+    # The cartesian coordinates should be pre-allocated....
+    p = Gv_to_array(Gplusk_vectors_cart(op.basis, op.kpoint))
     for α = 1:3
         iszero(op.Apot[α]) && continue
-        pα = [Gk[α] for Gk in Gplusk_vectors_cart(op.basis, op.kpoint)]
-        ∂αψ_fourier = pα .* ψ.fourier
+        ∂αψ_fourier = p[α, :] .* ψ.fourier
         ∂αψ_real = ifft(op.basis, op.kpoint, ∂αψ_fourier)
         Hψ.real .+= op.Apot[α] .* ∂αψ_real
     end
 end
+
+function Gv_to_array(Gv::AbstractArray{SVector{3, T}}) where {T}
+    reinterpret(reshape, T, Gv)
+end
```

Comment:
Why do you need this as a separate function?

Reply:
I put it as a separate function because it may be useful for other things as well. Although at the moment it is only used in `apply!(::MagneticFileOperator`. Putting it as a function also makes the type interface possible as the `T` needs to be inferred from the input.